service/src/main/java/bio/terra/pipelines/service/PipelineInputsOutputsService.java

```diff
@@ -87,22 +90,44 @@ public Map<String, Map<String, String>> prepareFileInputs(
     for (String fileInputName : fileInputNames) {
       String fileInputValue = (String) userProvidedInputs.get(fileInputName);
       String objectName = constructDestinationBlobNameForUserInputFile(jobId, fileInputValue);
-      String signedUrl =
-          gcsService.generatePutObjectSignedUrl(googleProjectId, bucketName, objectName).toString();
+      String signedUrl = getSignedUrl(googleProjectId, bucketName, objectName, useResumableUploads);
 
       fileInputsMap.put(
           fileInputName,
           Map.of(
               "signedUrl",
               signedUrl,
               "curlCommand",
-              "curl --progress-bar -X PUT -H 'Content-Type: application/octet-stream' --upload-file %s '%s' | cat"
-                  .formatted(fileInputValue, signedUrl)));
+              getCurlCommand(fileInputValue, signedUrl, useResumableUploads)));
     }
 
     return fileInputsMap;
   }
 
+  private String getSignedUrl(
+      String googleProjectId, String bucketName, String objectName, Boolean useResumableUploads) {
+    if (useResumableUploads) {
+      return gcsService
+          .generateResumablePostObjectSignedUrl(googleProjectId, bucketName, objectName)
+          .toString();
+    } else {
+      return gcsService
+          .generatePutObjectSignedUrl(googleProjectId, bucketName, objectName)
+          .toString();
+    }
+  }
+
+  private String getCurlCommand(
+      String fileInputValue, String signedUrl, Boolean useResumableUploads) {
+    if (useResumableUploads) {
+      return "curl --progress-bar -X PUT -H 'Content-Type: application/octet-stream' --upload-file %s $(curl -s -i -X POST -H 'x-goog-resumable: start' '%s' | grep -i '^Location:' | cut -d' ' -f2- | tr -d '\r') | cat"
```

Comment:
this should work as a drop-in replacement for the existing single-request curl. this command will initiate a resumable upload session by exchanging the signed url for destination url, and then it will perform the upload. 

obviously there isn't any retry handling but the idea is if someone gets their hands on this, it'll work fine

Reply:
does it fail if someone uses the original curl command with a resumable signed url? also this is a crazy curl command lol

Reply:
you are a good person for figuring out this command